main.go

```diff
@@ -80,48 +30,33 @@ func main() {
 }
 
 func realMain() error {
-	cfg, err := parseConfig(os.Args[1:])
+	p, m, err := parseConfig(os.Args[1:])
 	if err != nil {
 		if err == flag.ErrHelp {
 			return nil
 		}
 		return err
 	}
 
-	err = cfg.validate()
-	if err != nil {
-		return err
-	}
-
-	node, err := cfg.parse()
-	if err != nil {
-		return err
-	}
-
-	start, end, err := cfg.findSelection(node)
-	if err != nil {
-		return err
-	}
-
-	rewrittenNode, errs := cfg.rewrite(node, start, end)
-	if errs != nil {
-		if _, ok := errs.(*rewriteErrors); !ok {
-			return errs
-		}
-	}
+	return run(p, m)
+}
 
-	out, err := cfg.format(rewrittenNode, errs)
-	if err != nil {
-		return err
-	}
+// structType contains a structType node and it's name. It's a convenient
+// helper type, because *ast.StructType doesn't contain the name of the struct
+type structType struct {
+	name string
+	node *ast.StructType
+}
 
-	if !cfg.quiet {
-		fmt.Println(out)
-	}
-	return nil
+// output is used usually by editors
+type output struct {
+	Start  int      `json:"start"`
+	End    int      `json:"end"`
+	Lines  []string `json:"lines"`
+	Errors []string `json:"errors,omitempty"`
 }
 
-func parseConfig(args []string) (*config, error) {
+func parseConfig(args []string) (*pass, *modifytags.Mutation, error) {
```

Comment:
How about calling this 'parseFlags', since even if we revert 'pass' to 'config', this function still does more than just create the config.